### ISSUE TITLE
Abnormal Closure - code 1006

### ISSUE THREAD

**Issue by yvesroos** [tidwall/SwiftWebSocket#3]
After a few minutes he closes the connection giving the error "Abnormal Closure" (code 1006), someone knows what can be?
When I used socket rocket in objective-c worked

**Comment by tidwall** [comment]
From the [RFC 6455](http://tools.ietf.org/html/rfc6455#section-7.4.1)

```
1006 is a reserved value and MUST NOT be set as a status code in a
Close control frame by an endpoint.  It is designated for use in
applications expecting a status code to indicate that the
connection was closed abnormally, e.g., without sending or
receiving a Close control frame.
```

SwiftWebSocket follows this rule when the underlying socket is closed without a close frame sent from the server.  

Occasionally a websocket server requires that the client sends occasional messages to keep the connection alive; do you think that this might be the cause?

Could you post the code that is failing?

**Comment by tidwall** [comment]
I'm closing this issue for now. Please feel free to reopen it if this issue is reproducible.

**Comment by zentrope** [comment]
Just as a data point, I'm also getting this.

I have a server written in Clojure using the Aleph web server. I also have a web client which never times out like this. I don't think it's the server side.

Is this something related to iOS itself?

My code is about the same as on your README. It makes a connection, then waits for data from the server, and that's it. Data flows through okay, but if I don't do anything, the abnormal close happens.

Using the swift/1.2 branch.

**Comment by tidwall** [comment]
It doesn't sound like an iOS related issue. Does the client send data to the server, or is it only server to client?

**Comment by zentrope** [comment]
Eventually it's supposed to be a two-way system, but it's chat (basically). If nothing's happening, no data will cross in either direction.

I just set up an NSTimer to send "ping" to the server every 30 seconds. Got two of them out before the 1006 error happened again.

**Comment by zentrope** [comment]
This is happening on the simulator, by the way.

**Comment by zentrope** [comment]
If I have the client send "ping" messages, the connection will eventually drop. About 1 minute, I think.

If I have the client send a "ping" message, and the server always respond with a "ping" message, the connection holds.

**Comment by zentrope** [comment]
https://gist.github.com/zentrope/e747a8419e58c821fa11

**Comment by zentrope** [comment]
Okay, one last data point. I switched the code to [Starscream](https://github.com/daltoniam/Starscream), and I don't get the timeout behavior. That suggests that iOS (the simulator, anyway) nor the server is the problem.

**Comment by tidwall** [comment]
This is very helpful. It definitely sounds like an issue with the client. I'll see what I can do.

**Comment by tidwall** [comment]
I pushed a fix to the swift/1.2 branch. Thanks for your help.
